tutorials/20_score_based_methods_new_features.ipynb

```diff
@@ -0,0 +1,710 @@
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "Drawing 10000 posterior samples: 100%|██████████| 499/499 [00:17<00:00, 28.73it/s]\n",
+      "\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "\u001b[A\n",
+      "Drawing 100 posterior samples: 100%|██████████| 499/499 [00:03<00:00, 148.17it/s]\n",
+      "Drawing 10000 posterior samples: 10099it [00:20, 486.58it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Build the posterior using the trained score estimate, and sample with the sde option\n",
+    "posterior = inference.build_posterior(score_estimator, sample_with=\"sde\")\n",
+    "samples = posterior.sample((10000,), x=x_o, predictor=\"euler_maruyama\", corrector=\"langevin\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The argument `sample_with` allows to choose which kind of solver we want to use to reverse the diffusion equation :\n",
+    "\n",
+    "- `\"ode\"` : builds the probability flow ODE using the zuko library. We can sample from the flow afterwards by calling `.sample()`\n",
+    "\n",
+    "- `\"sde\"` : solves the reverse SDE by alternating **prediction** and **correction** steps  \n",
+    "\n",
+    "Both options contain **rejection sampling** steps to ensure prior coverage.  \n",
+    "Note that in the `\"sde\"` option, the only available predictor is `\"euler_maruyama\"` and the available correctors are `\"langevin\"` (for **Unadjusted Langevin Dynamics**) and `\"gibbs\"`.  \n",
+    "\n",
+    "The argument `x_o` in the `.sample()` method indicates which observation the posterior should be evaluated at before sampling and replaces the `.set_default()`method."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### MAP estimation\n",
+    "It is possible to get a Maximum A Posterior (MAP) estimate in the case of score-based methods. Note that in this case, you should **first** evaluate the posterior at the desired observation, using the `set_default_x` method !  \n",
```

Comment:
edit: we are not really evaluating the posterior by setting `set_default_x`, we just setting the "default" `x` that we will then when calling MAP or sample.